lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4323,25 +4325,36 @@ namespace {
       // Check whether there is a function with the same name as this
       // property. If so, suppress the property; the user will have to use
       // the methods directly, to avoid ambiguities.
-      auto containerTy = dc->getDeclaredTypeInContext();
+      Type containerTy;
+      if (auto *classDecl = dyn_cast<ClassDecl>(dc)) {
+        // If we're importing into the primary @interface for something, as
+        // opposed to an extension, make sure we don't try to load any
+        // categories...by just looking into the super type.
+        containerTy = classDecl->getSuperclass();
```

Comment:
Can we rename `containerTy` to something more meaningful, now that it doesn't mean "the type into which this declaration will go". How about `lookupContainerTy`?

We don't need to look into the protocols to which this class conforms, because it's not an "override" in that case, right?